src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/TestExtensions.cs

```diff
@@ -390,4 +479,35 @@ internal void InvalidateCache()
         }
     }
 
+    private void AddCachedExtensionToExtensionMap<T>(
+        IDictionary<string, ISet<string>> extensionMap,
+        string extensionType,
+        IList<T> extensions) where T : TestPluginInformation
+    {
+        if (extensions == null) return;
+
+        extensionMap.Add(extensionType, new HashSet<string>());
+        foreach (var extension in extensions)
+        {
+            extensionMap[extensionType].Add(extension.IdentifierData);
+        }
+    }
+
+    private static string SerializeExtensionMap(IDictionary<string, ISet<string>> extensionMap)
+    {
+        StringBuilder sb = new();
+
+        foreach (var kvp in extensionMap)
+        {
+            sb.Append(
```

Comment:
I would refactor to add a if condition around and use `AppendFormat`.

Reply:
Done.